rerun_py/rerun_sdk/rerun/error_utils.py

```diff
@@ -32,11 +81,120 @@ def _send_warning(
     from rerun._log import log
     from rerun.archetypes import TextLog
 
-    if rerun.strict_mode():
+    if check_strict_mode():
         raise TypeError(message)
 
-    context_descriptor = _build_warning_context_string(skip_first=depth_to_user_code + 2)
-    warning = f"{message}\n{context_descriptor}"
+    # Send the warning to the user first
+    warnings.warn(message, category=RerunWarning, stacklevel=depth_to_user_code + 1)
+
+    # Logging the warning to Rerun is a complex operation could produce another warning. Avoid recursion.
+    if not getattr(_rerun_exception_ctx, "sending_warning", False):
+        _rerun_exception_ctx.sending_warning = True
+
+        # TODO(jleibs): Context/stack should be its own component.
+        context_descriptor = _build_warning_context_string(skip_first=depth_to_user_code + 1)
+
+        log("rerun", TextLog(text=f"{message}\n{context_descriptor}", level="WARN"), recording=recording)
```

Comment:
we should document this special entity name `rerun` somewhere

Reply:
https://github.com/rerun-io/rerun/issues/3571